Calculator/Calculator/Controller/ViewController.swift

```diff
@@ -1,15 +1,201 @@
 //
 //  Calculator - ViewController.swift
-//  Created by yagom. 
-//  Copyright © yagom. All rights reserved.
-// 
+//  Created by DuDu
+//
 
 import UIKit
 
-class ViewController: UIViewController {
-
+final class ViewController: UIViewController {
+    private var currentNumber: String = ""
+    private var expression: [String?] = []
+    private var isInputExist: Bool = false
+    private var isCalculateValue: Bool = false
+    
+    @IBOutlet private var logScrollView: UIScrollView!
+    @IBOutlet private var calculateLogStackView: UIStackView!
+    
+    @IBOutlet private var numberLabel: UILabel!
+    @IBOutlet private var operatorLabel: UILabel!
+    
+    private lazy var numberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .decimal
+        formatter.maximumFractionDigits = 20
+        formatter.roundingMode = .floor
+        return formatter
+    }()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        numberLabel.text = "0"
+        logScrollView.showsVerticalScrollIndicator = false
+    }
+    
+    @IBAction private func numberButtonDidTapped(_ sender: UIButton) {
+        if numberLabel.text == "NaN" {
+            return
+        }
+        
+        if String(currentNumber).count >= 20 {
+            return
+        }
+        
+        if isCalculateValue {
+            isCalculateValue = false
+            acButtonDidTapped(nil)
+        }
+        
+        guard let inputNumber = sender.titleLabel?.text else {
+            return
+        }
+        
+        if ["00","0"].contains(inputNumber), numberLabel.text == "0" {
```

Comment:
요거 반대가 되어야하는거 아닐까용..? 
```suggestion
        if inputNumber.contains(["00","0"]), numberLabel.text == "0" {
```

Reply:
반대로 하면 컴파일 에러가 나지 않나요..?

해당코드는, 현재 계산기의 numberLabel.text가 0인데, 0이나, 00을 입력받았을때 
000000 이런식으로 써줄 필요가 없잖아요? (0은 0이니깐)
그부분을 거르기 위한 코드였습니다..!

혹시 제가 이해를 잘 못하고 있다면 바로 말씀해주세요..!
이번프로젝트에서 side effect 관리에 아예 실패해버려서 너무 힘듭니다 ㅠㅠ

Reply:
아핫 이해했습니다 ㅎㅎ 저도 보면서 헷갈렷나보네요